corsikaio/subblocks/run_header.py

```diff
@@ -79,10 +79,17 @@
 
 
 def warn():
-    warnings.warn('Version unknown, using run header definition of version 7.x')
+    warnings.warn("Version unknown, using run header definition of version 7.x")
     return run_header_dtype_7x
 
 
+run_header_fields = defaultdict(warn)
```

Comment:
Warn returns the dtype, you need two functions now. One for the fields, one for the dtypes